File: app/src/main/java/com/example/weatherapp/DataQuery.java
package com.example.weatherapp;

import android.util.Log;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class DataQuery {
    public static City fetchCityData(String cityQuery, String tempType) {
        City city = null;

        String JSON_WEATHER_URL = "https://api.openweathermap.org/data/2.5/weather?q=";
        String TYPE = "&units=" + (tempType.equals("C") ? "metric" : "imperial");
        String URL_TAIL = "&appid=cf7ef9156622ecccc8decb4a00a549b1";
        try {
            String stringUrl = JSON_WEATHER_URL+cityQuery+TYPE+URL_TAIL;
            URL url = new URL(stringUrl);
            HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
            httpURLConnection.setRequestMethod("GET");
            httpURLConnection.connect();

            if (httpURLConnection.getResponseCode() == HttpURLConnection.HTTP_OK) {
                InputStream inputStream = httpURLConnection.getInputStream();
                String jsonResponse = readFromStream(inputStream);
                JSONObject baseJsonResponse = new JSONObject(jsonResponse);

                JSONObject tempMain = baseJsonResponse.getJSONObject("main");
                Double temp = tempMain.getDouble("temp");
                String name = baseJsonResponse.getString("name");

                city = new City(name,tempType,temp);
            }
        } catch (Exception e) {
            Log.e("fetchCityData", "Problem parsing the city JSON results", e);
        }

        return city;
    }

    private static String readFromStream(InputStream inputStream) throws IOException {
        StringBuilder output = new StringBuilder();
        if (inputStream != null) {
            InputStreamReader inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
            BufferedReader reader = new BufferedReader(inputStreamReader);
            String line = reader.readLine();
            while (line != null) {
                output.append(line);
                line = reader.readLine();
            }
        }
        return output.toString();
    }

}

File: app/src/main/java/com/example/weatherapp/DataQuery.kt
package com.example.weatherapp

import android.util.Log
import org.json.JSONObject
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL
import java.nio.charset.StandardCharsets

object DataQuery {
    @JvmStatic
    fun fetchCityData(cityQuery: String): City? {
        var city: City? = null
        val JSON_WEATHER_URL = "https://api.openweathermap.org/data/2.5/weather?q="
        val URL_TAIL = "&appid=cf7ef9156622ecccc8decb4a00a549b1"
        try {
            val stringUrl = JSON_WEATHER_URL + cityQuery + URL_TAIL
            val url = URL(stringUrl)
            val httpURLConnection = url.openConnection() as HttpURLConnection
            httpURLConnection.requestMethod = "GET"
            httpURLConnection.connect()
            if (httpURLConnection.responseCode == HttpURLConnection.HTTP_OK) {
                val inputStream = httpURLConnection.inputStream
                val jsonResponse = readFromStream(inputStream)
                val baseJsonResponse = JSONObject(jsonResponse)
                val tempMain = baseJsonResponse.getJSONObject("main")
                val temp = tempMain.getDouble("temp")
                val name = baseJsonResponse.getString("name")
                city = City(name, temp)
            }
        } catch (e: Exception) {
            Log.e("fetchCityData", "Problem parsing the city JSON results", e)
        }
        return city
    }

    @Throws(IOException::class)
    private fun readFromStream(inputStream: InputStream?): String {
        val output = StringBuilder()
        if (inputStream != null) {
            val inputStreamReader = InputStreamReader(inputStream, StandardCharsets.UTF_8)
            val reader = BufferedReader(inputStreamReader)
            var line = reader.readLine()
            while (line != null) {
                output.append(line)
                line = reader.readLine()
            }
        }
        return output.toString()
    }
}